src/pages/publication/FilesAndLicensePanel.tsx

```diff
@@ -44,17 +44,23 @@ const FilesAndLicensePanel: React.FC<FilesAndLicensePanelProps> = ({ goToNextTab
   const { t } = useTranslation('publication');
   const { values, setFieldTouched }: FormikProps<FormikPublication> = useFormikContext();
   const [isLicenseModalOpen, setIsLicenseModalOpen] = useState(false);
+  const {
+    fileSet,
+    entityDescription: {
+      reference: { publicationContext },
+    },
+  } = values;
 
-  const valuesRef = useRef(values);
+  const filesLengthRef = useRef(fileSet.length);
   useEffect(() => {
-    valuesRef.current = values;
-  }, [values]);
+    filesLengthRef.current = fileSet.length;
+  }, [fileSet.length]);
 
   // Set all fields to touched on unmount
   useEffect(() => {
     return () => {
-      // Use valuesRef to avoid trigging this useEffect on every values update
-      const fieldNames = getAllFileFields(valuesRef.current.fileSet.length);
```

Comment:
nå synes jeg det ble enda mer klart at det er unødvendig med useRef og useEffect.

Kan du ikke bare slette ref'en og useEffecten over, og ha dette i denne useEffecten?
```
  // Set all fields to touched on unmount
  useEffect(() => {
    return () => {
      const fieldNames = getAllFileFields(fileSet.length);
      fieldNames.forEach((fieldName) => setFieldTouched(fieldName));
    };
  }, [setFieldTouched, fileSet.length]);
```

Reply:
hvis man da laster opp to filer vil den andre fila føre til at `fileSet.length` endrer seg, som trigger useEffecten, og feltene blir satt til touched selv om de ikke er rørt
validering av FieldArray er litt kinkig :p

Reply:
ser den ja, men skjer ikke dette på onUnmount?

Reply:
ah, tok ned branchen nå for å teste. skjønner hva du mener.

Reply:
det er det vi vil ja, men hvis `fileSet.length` er en dependency vil v1 av useEffecten returnere (som setter felter til touched), før den lager en v2 som venter på neste endring
ved å bruke useRef vil det i praksis være en god gammeldags componentWillUnmount (eller hva den het) sidenvi bare har `setFieldTouched` (som vi "vet" aldri vil endres) som dependency 
Gir det mening?

Reply:
det som faktisk vil fungere like godt er dette:
```
// Set all fields to touched on unmount
  useEffect(() => {
    return () => {
      const fieldNames = getAllFileFields(fileSet.length);
      fieldNames.forEach((fieldName) => setFieldTouched(fieldName));
    };
  }, [setFieldTouched]);
```
Men da klager lintinga på at det mangler en dependency. Kunne evt satt en disable warnings for akkurat den linja, men warnings er sikkert warnings for en grunn også :p

Reply:
hehe, nei, er ikke så lurt å disable warnings 😅bedre å gjøre det sånn du tenkte